common/src/main/java/io/druid/metadata/DefaultPasswordProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2015 - Yahoo! Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.metadata;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class DefaultPasswordProvider implements PasswordProvider
+{
+  private final String password;
+
+  @JsonCreator
+  public static DefaultPasswordProvider fromString(String str)
+  {
+    return new DefaultPasswordProvider(str);
+  }
+
+  @JsonCreator
+  public DefaultPasswordProvider(@JsonProperty("password") String password)
+  {
+    this.password = password;
+  }
+
+  @Override
+  @JsonProperty
+  public String getPassword()
+  {
+    return password;
+  }
+
+  @Override
+  public String toString() {
```

Comment:
we currently just print out '*****' regardless of the size of the password, I don't think there is a need to print out anything special. Even printing out a salt/hash could be a security problem.